### PR TITLE
Reduces the size of the DB API to make it easier to implement. Some clean-up.

### PR DESCRIPTION
**Pull request by joeldudley**

### CODE REVIEW

File: go/obscuronode/enclave/db.go [hunk 1]
```diff
@@ -13,45 +12,41 @@ import (
...
-	FetchGossipedRollups(height int) []*Rollup
+	RollupResolver
 
 	// Block resolver
-	HeightBlock(block *types.Block) int
-	ParentBlock(block *types.Block) (*types.Block, bool)
-	ResolveBlock(hash obscurocommon.L1RootHash) (*types.Block, bool)
-	StoreBlock(node *types.Block)
+	BlockResolver
+
```
[the remark this answers is not in the document]
Comment (joeldudley) [reply go/obscuronode/enclave/db.go:33]: On closer inspection, `RollupResolver` had no reason to exist (unlike `BlockResolver`, which is sometimes used as an argument to not expose all the DB methods). Deleted.

**Review by otherview — COMMENTED** [review]
The PR looks good - I think there might be a bit of duplication between the storage and the DB, the line between the two seem blurry. Something that will get clearer with time.

That being said it's way more polished than it was :)
If we could add comment to the methods I'd be happy to merge it in.

**Comment by tudor-malene** [comment]
Thanks, @joeldudley, can you create a small list of each change and a short reason for it?
It's a bit difficult to figure it out from the diff alone.
Cheers

**Comment by joeldudley** [comment]
@tudor-malene Sure:

* Split `db.go` into `db.go` and `storage.go`. Previously, `DB` contained a lot of logic that wouldn't normally live in the DB (e.g. looping over stuff, forking). The idea was to slim down `DB` to a minimal set of methods that map more closely to actual DB methods, then add helper stuff to `Storage` (which wraps `DB` - the user won't see `DB`). In turn, the idea was that when we back `DB` with an actual datastore, it would be easier
* Embed the `BlockResolver` interface directly in `DB` instead of duplicating it
* General clean-up, especially better grouping of methods in `DB` interface, docstrings explaining the behaviour of each, and consistent naming

**Review by otherview — APPROVED** [review]
I think it's good. 
We should probably do other stuff such as removing business logic from the `enclave/db.go` ( Like the StoreRollup method), and move the  `enclave/db.go` outside the enclave package because it's should be agnostic to the enclave operations. But we don't need to do that now imo.

**Pull request merged**